### ISSUE TITLE
registerHandler not working with ValueNotifier

### ISSUE THREAD

**Issue by martyfuhry** [escamoteur/watch_it#28]
If I have a local `ValueNotifier` and attempt to use `registerHandler`, I get an error:

> Invalid argument(s): Either the return type of the select function or the type T has to be a Listenable

This tracks down to the following check in `registerHandler`:

```dart
    if (T is Listenable) {
      observedObject = (parentObject) as Listenable;
    } else {
      throw ArgumentError(
          'Either the return type of the select function or the type T has to be a Listenable');
    }
```

I'm using it like this:

```dart
class MyWidget extends WatchingStatefulWidget {
...
}

class _MyWidgetState extends State<MyWidget> {
  final ValueNotifier<int> _myValueNotifier = ValueNotifier(0);

  @override
  Widget build(BuildContext context) {
    registerHandler(
      target: _myValueNotifier,
      handler: (context, value, cancel) {
        print('value is $value');
      }
    );
    ...
  }
}
```

But this is probably the wrong thing to check. In this case, 

```dart
print(ValueNotifier is Listenable); // prints false
print(_myValueNotifier is Listenable); // prints true
```

I have no earthly idea why `ValueNotifier is Listenable` evaluates to false, but it certainly does.

**Comment by escamoteur** [comment]
ok, the reason is explained here, I have to fix this I too was expecting it to work that way

https://github.com/dart-lang/sdk/issues/43390

**Comment by escamoteur** [comment]
fixed in V1.4.1
